gpii/node_modules/testing/src/RunTestDefs.js

```diff
@@ -0,0 +1,240 @@
+    var configurationName = testDef.configType || kettle.config.createDefaults(testDef.config);
+    //var caseHolderOptions = fluid.filterKeys(testDef, ["modules", "sequence", "config"], true);
+    var caseHolderOptions = fluid.filterKeys(testDef, ["modules", "sequence"], true);
+    caseHolderOptions.modules = [{
+        name: configurationName + " tests",
+        tests: [{
+            name: testDef.name,
+            expect: testDef.expect,
+            sequence: testDef.sequence,
+            sequenceGrade: testDef.sequenceGrade || sequenceGrade
+        }]
+    }];
+
+    return {
+        type: environmentType,
+        options: {
+            configurationName: configurationName,
+            components: {
+                tests: {
+                    options: caseHolderOptions
+                }
+            }
+        }
+    };
+};
+
+gpii.test.testDefToServerEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.serverEnvironment", "gpii.test.standardServerSequenceGrade");
+};
+
+gpii.test.testDefToCouchEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", "gpii.test.couchSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", "gpii.test.snapshotSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotWithInitialStateEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", "gpii.test.initialStateSequenceGrade");
+};
+
+/*
+
+
+gpii.test.testDefToServerEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.serverEnvironment", testDef.sequenceGrade || "gpii.test.standardServerSequenceGrade");
+};
+
+gpii.test.testDefToCouchEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", testDef.sequenceGrade || "gpii.test.couchSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", testDef.sequenceGrade || "gpii.test.snapshotSequenceGrade");
+};
+
+gpii.test.testDefToSnapshotWithInitialStateEnvironment = function (testDef) {
+    return gpii.test.testDefToEnvironment(testDef, "gpii.test.couchEnvironment", testDef.sequenceGrade || "gpii.test.initialStateSequenceGrade");
+};
+
+ */
+
+
+/* Build a test fixture for integration/acceptance tests operating the stereotypical workflow -
+ * snapshot, login, expectConfigured, logout and expectRestored
+ */
+gpii.test.buildSingleTestFixture = function (testDef, rootGrades) {
+    testDef.gradeNames = fluid.makeArray(testDef.gradeNames).concat(fluid.makeArray(rootGrades));
+
+    // TODO: Discuss whether this is useful/advisable.
+    testDef.expect = testDef.initialState ? 6 : 5;
```

Comment:
Not really, "expects" should be carried along with the sequence grades

Reply:
OK, I'll try removing it.